galasa-parent/dev.galasa.framework.api.authentication/src/main/java/dev/galasa/framework/api/authentication/internal/OidcProvider.java

```diff
@@ -359,4 +369,29 @@ private void refreshJsonWebKeys() throws IOException, InterruptedException {
         // Update the next refresh time by the refresh interval
         nextJwkRefresh = Instant.now().plus(JWK_REFRESH_INTERVAL_MINUTES, ChronoUnit.MINUTES);
     }
+
+    /**
+     * Validates and returns the validated URL as a string. The validation checks that
+     * the given URL is a valid URL and its scheme and host matches the OpenID Connect
+     * issuer's scheme and host.
+     * 
+     * @param urlToValidate the URL to validate
+     * @return the validated URL
+     * @throws ServletException if the URL is not valid
+     */
+    private String getValidatedUrl(String urlToValidate) throws ServletException {
+        try {
+            URI uri = new URI(urlToValidate);
+            if (!uri.getScheme().equals(issuerUrl.getScheme())
+                || !uri.getHost().equals(issuerUrl.getHost())) {
+                ServletError error = new ServletError(GAL5061_MISMATCHED_OIDC_URI_RECEIVED, urlToValidate, issuerUrl.toString());
```

Comment:
same here

Reply:
Addressed in https://github.com/galasa-dev/framework/pull/566/commits/8c56f5c11bf27c9e99ea1f836408149fac1295a3